### PR TITLE
update low level api to book, ses-1

### PR DESCRIPTION
**Pull request by junjun315**
Modify chapter-1 code from high level API to low level API

### CODE REVIEW

File: 01.fit_a_line/README.cn.md [hunk 1]
```diff
@@ -173,112 +155,137 @@ use_cuda = False
...
-    optimizer_func=optimizer_program)
+# Plot data
+from paddle.v2.plot import Ploter
+
+train_title = "Train cost"
+test_title = "Test cost"
+plot_cost = Ploter(train_title, test_title)
+
+
+def event_handler(title, loop_step, handler_val):
```
Comment (reyoung) [inline 01.fit_a_line/README.cn.md]: Event Handler is not needed when using low-level APIs.

You can just plot/print in-placely.

Comment (junjun315) [reply 01.fit_a_line/README.cn.md]: ok, thanks review!

Comment (junjun315) [reply 01.fit_a_line/README.cn.md]: Has been updated at 3d0283e

File: 01.fit_a_line/README.cn.md [hunk 2]
```diff
@@ -173,112 +155,135 @@ use_cuda = False
...
-训练器会读入一个训练程序和一些必要的其他参数：
+除此之外，还可以通过画图，来展现`训练进程`：
 
 ```python
-trainer = Trainer(
-    train_func=train_program,
-    place=place,
-    optimizer_func=optimizer_program)
+# Plot data
+from paddle.v2.plot import Ploter
```
Comment (luotao1) [inline 01.fit_a_line/README.cn.md]: 这个画图的Ploter，为什么在v2下面呢？from paddle.utils import Ploter可以用么？
https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/utils/__init__.py#L15

File: 01.fit_a_line/README.cn.md [hunk 3]
```diff
@@ -123,7 +115,7 @@ except ImportError:
 1. 数据下载的过程。下载数据保存在~/.cache/paddle/dataset/uci_housing/housing.data。
 2. [数据预处理](#数据预处理)的过程。
 
-接下来我们定义了用于训练和测试的数据提供器。提供器每次读入一个大小为`BATCH_SIZE`的数据批次。如果用户希望加一些随机性，她可以同时定义一个批次大小和一个缓存大小。这样的话，每次数据提供器会从缓存中随机读取批次大小那么多的数据。
+接下来我们定义了用于训练的数据提供器。提供器每次读入一个大小为`BATCH_SIZE`的数据批次。如果用户希望加一些随机性，她可以同时定义一个批次大小和一个缓存大小。这样的话，每次数据提供器会从缓存中随机读取批次大小那么多的数据。
```
Comment (luotao1) [inline 01.fit_a_line/README.cn.md]: 这里为什么是“她”？应该是“它”吧。

File: 01.fit_a_line/train.py [hunk 4]
```diff
@@ -13,122 +13,135 @@
...
-                # If the accuracy is good enough, we can stop the training.
-                print('loss is less than 10.0, stop')
-                trainer.stop()
-        step += 1
-
-    if isinstance(event, EndEpochEvent):
-        if event.epoch % 10 == 0:
-            # We can save the trained parameters for the inferences later
+    main_program = fluid.default_main_program()
+    star_program = fluid.default_startup_program()
```
Comment (reyoung) [inline 01.fit_a_line/train.py]: `star_program` --> `startup_program`

File: 01.fit_a_line/train.py [hunk 5]
```diff
@@ -13,122 +13,135 @@
...
+    use_cuda = False
+    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+
+    # Specify the directory to save the parameters
+    params_dirname = "fit_a_line.inference.model"
+    num_epochs = 100
+
+    # For training test cost
+    def train_test(program, feeder):
```
Comment (reyoung) [inline 01.fit_a_line/train.py]: It seems that `train_test` is not necessary. Perhaps we can inline this function.

File: 01.fit_a_line/README.cn.md [hunk 6]
```diff
@@ -173,112 +155,135 @@ use_cuda = False
...
-feed_order=['x', 'y']
+exe = fluid.Executor(place)
+num_epochs = 100
+
+# For training test cost
+def train_test(train_program, feeder):
+    exe_test = fluid.Executor(place)
+    accumulated = 1 * [0]
+    count = 0
+    test_program = train_program.clone(for_test=True)
```
Comment (reyoung) [inline 01.fit_a_line/README.cn.md]: These lines between markdown and source file are different.

File: 01.fit_a_line/train.py [hunk 7]
```diff
@@ -13,122 +13,135 @@
...
+                program=program,
+                feed=feeder.feed(data_test),
+                fetch_list=[avg_loss])
+            accumulated = [x_c[0] + x_c[1][0] for x_c in zip(accumulated, outs)]
+            count += 1
+        return [x_d / count for x_d in accumulated]
+
+    # main train loop.
+    def train_loop():
+
```
Comment (reyoung) [inline 01.fit_a_line/train.py]: unused empty line

File: 01.fit_a_line/train.py [hunk 8]
```diff
@@ -13,122 +13,135 @@
...
+        feeder = fluid.DataFeeder(place=place, feed_list=[x, y])
+        feeder_test = fluid.DataFeeder(place=place, feed_list=[x, y])
```
Comment (reyoung) [inline 01.fit_a_line/train.py]: It seems that we can use the same feeder for both train and test.

File: 01.fit_a_line/train.py [hunk 9]
```diff
@@ -13,122 +13,135 @@
...
+        exe.run(star_program)
+
+        train_title = "Train cost"
```
Comment (reyoung) [inline 01.fit_a_line/train.py]: `train_title` --> `train_prompt`

File: 01.fit_a_line/train.py [hunk 10]
```diff
@@ -13,122 +13,135 @@
...
+
+                if math.isnan(float(avg_loss_value)):
+                    sys.exit("got NaN loss, training failed.")
             if params_dirname is not None:
-                trainer.save_params(params_dirname)
+                # We can save the trained parameters for the inferences later
+                fluid.io.save_inference_model(params_dirname, ['x'],
+                                              [y_predict], exe)
 
+    train_loop()
```
Comment (reyoung) [inline 01.fit_a_line/train.py]: `train_loop` is not needed. We can in-place it right here.

File: 01.fit_a_line/train.py [hunk 11]
```diff
@@ -13,122 +13,135 @@
...
+    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+
+    # Specify the directory to save the parameters
+    params_dirname = "fit_a_line.inference.model"
+    num_epochs = 100
+
+    # For training test cost
+    def train_test(program, feeder):
+        exe_test = fluid.Executor(place)
```
Comment (reyoung) [inline 01.fit_a_line/train.py]: It is better to use ParallelExecutor rather than Executor for training and testing.

File: 01.fit_a_line/train.py [hunk 12]
```diff
@@ -13,122 +13,135 @@
...
-
-train_reader = paddle.batch(
-    paddle.reader.shuffle(paddle.dataset.uci_housing.train(), buf_size=500),
-    batch_size=BATCH_SIZE)
-
-test_reader = paddle.batch(
-    paddle.reader.shuffle(paddle.dataset.uci_housing.test(), buf_size=500),
-    batch_size=BATCH_SIZE)
 
+def main():
```
Comment (reyoung) [inline 01.fit_a_line/train.py:35]: Unused empty line.

Comment (junjun315) [reply 01.fit_a_line/train.py:35]: thanks, had Deleted

File: 01.fit_a_line/train.py (hunk 11 above)
Comment (junjun315) [reply 01.fit_a_line/train.py]: have changed to ParallelExecutor～

File: 01.fit_a_line/train.py (hunk 10 above)
Comment (junjun315) [reply 01.fit_a_line/train.py]: ok，done

File: 01.fit_a_line/train.py (hunk 9 above)
Comment (junjun315) [reply 01.fit_a_line/train.py]: ok

File: 01.fit_a_line/train.py (hunk 8 above)
Comment (junjun315) [reply 01.fit_a_line/train.py]: ok，use feeder for both

File: 01.fit_a_line/README.cn.md (hunk 6 above)
Comment (junjun315) [reply 01.fit_a_line/README.cn.md]: fixed

File: 01.fit_a_line/train.py (hunk 5 above)
Comment (junjun315) [reply 01.fit_a_line/train.py]: move this `train_test ` out of `main`

File: 01.fit_a_line/README.cn.md (hunk 2 above)
Comment (junjun315) [reply 01.fit_a_line/README.cn.md]: 因为当前book对应镜像paddle版本较低，所以暂时不支持utils下的ploter,本次将升级镜像paddle版本并改为utils画图函数

**Review by luotao1 — APPROVED** [review]
LGTM
